### PR TITLE
sync-diff-inspector: let user router rule cover the default rule

### LINKED ISSUE #558: sync-diff-inspector: mysql to mysql check error
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

Create two databases and tables in one MySQL instance.

```
CREATE DATABASE inspector_mysql_0;
CREATE TABLE inspector_mysql_0.tb_emp1 (id INT(11), name VARCHAR(25), deptId INT(11));
INSERT INTO inspector_mysql_0.tb_emp1 values (1,"hello1",1);
CREATE DATABASE inspector_mysql_1;
CREATE TABLE inspector_mysql_1.tb_emp1 (id INT(11), name VARCHAR(25), deptId INT(11));
INSERT INTO inspector_mysql_1.tb_emp1 values (1,"hello1",1);
```

then run sync-diff-inspector with config.toml

```toml
# Diff Configuration.

######################### Global config #########################

# how many goroutines are created to check data
check-thread-count = 4

# set false if just want compare data by checksum, will skip select data when checksum is not equal.
# set true if want compare all different rows, will slow down the total compare time.
export-fix-sql = true

# ignore check table's data
check-struct-only = false


######################### Databases config #########################
[data-sources.mysql1]
    host = "127.0.0.1"
    port = 3306
    user = "root"
    password = "12344321"
    route-rules = ["rule1"]

[data-sources.tidb0]
    host = "127.0.0.1"
    port = 3306
    user = "root"
    password = "12344321"
    # remove comment if use tidb's snapshot data

[routes.rule1]
schema-pattern = "inspector_mysql_0"      # schema to match. Support wildcard characters * and ?.
table-pattern = "tb_emp1"          # table to match. Support wildcard characters * and ?.
target-schema = "inspector_mysql_1"         # target schema
target-table = "tb_emp1"             # target table

######################### Task config #########################
# Required
[task]
    # 1 fix sql: fix-target-TIDB1.sql
    # 2 log: sync-diff.log
    # 3 summary: summary.txt
    # 4 checkpoint: a dir
    output-dir = "/tmp/output/config"

    source-instances = ["mysql1"]

    target-instance = "tidb0"

    # tables need to check. *Include `schema` and `table`. Use `.` to split*
    target-check-tables = ["inspector_mysql_1.tb_emp1"]

    # extra table config
#     target-configs= []
    target-configs= ["config1"]

# Optional
# [table-configs]
[table-configs.config1]

# tables need to use this specified config.
# if use this config. target-tables should be a subset of #target-check-tables
target-tables = ["inspector_mysql_1.tb_emp1"]

range = "id > 0"
index-fields = [""]
ignore-columns = ["",""]
chunk-size = 0
collation = ""
```

2. What did you expect to see?

```
The table structure and data in following tables are equivalent

`inspector_mysql_1`.`tb_emp1`
```

3. What did you see instead?

```
The table structure and data in following tables are equivalent


The following tables contains inconsistent data

+-------------------------------+--------------------+----------------+
|             TABLE             | STRUCTURE EQUALITY | DATA DIFF ROWS |
+-------------------------------+--------------------+----------------+
| `inspector_mysql_1`.`tb_emp1` | true               | +1/-0          |
+-------------------------------+--------------------+----------------+
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

MySQL 8.0.22

5. which tool are you using?

tidb-tools/sync-diff-inspector

6. what version of tool are you using (`pump -V` or `tidb-lightning -V` or `syncer -V`)?

```
App Name: sync_diff_inspector v2.0
Release Version: v5.2.2-113-g39e3eb4-dirty
Git Commit Hash: 39e3eb4bb9725bf714bd6c23a77de69852ea2a27
Git Branch: master
UTC Build Time: 2021-11-28 07:25:14
Go Version: go1.17
```

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close #558 

### What is changed and how it works?

The function [`router.Route`](https://github.com/pingcap/tidb-tools/blob/3f5c34ef7e62e63b3a3410fe2d77bcfd9bea2934/pkg/table-router/router.go#L185) returns itself when there is no matching, So even though the rule has been replaced or removed, the target table will be still matched by the source upstream. 

So we need to add a sheild rule, such as `target-schema`.`target-table` -> `_____`.`_____`, for the source if needed.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: sync_diff_inspector/source/source.go [hunk 1]
```diff
@@ -134,23 +134,48 @@ func NewSources(ctx context.Context, cfg *config.Config) (downstream Source, ups
...
+				// There is a user rule routing to `tableConfig.Schema`.`tableConfig.Table`
+				rules := d.Router.Match(tableConfig.Schema, tableConfig.Table)
+
+				if len(rules) == 0 {
+					// There is no self match in these user rules.
+					// Need to shield the table for this source.
+					if d.Router.AddRule(&router.TableRule{
+						SchemaPattern: tableConfig.Schema,
+						TablePattern:  tableConfig.Table,
+						TargetSchema:  "_____",
```
Comment (3pointer) [inline sync_diff_inspector/source/source.go]: maybe use other stranger name and make it const. user may have the table name like '_____'?

Comment (3pointer) [reply sync_diff_inspector/source/source.go]: image

**Comment by 3pointer** [comment]
rest LGTM

File: sync_diff_inspector/source/source.go (hunk 1 above)
Comment (Leavrth) [reply sync_diff_inspector/source/source.go]: done

**Pull request merged**